### DISCUSSION TITLE
Taking a child XmlElement, removing it and wrapping it around its parent

### DISCUSSION THREAD

**Discussion by BrayFlex** [renggli/dart-xml#168] [Q&A] [answered]
Lets say my XmlElement was based on this:

```xml
<XML>
    <BOX>
        <IMG>
            <BLUR />
            <URL value="./" />
        </IMG>
    </BOX>
</XML>
```

I want to modify it to remove the `BLUR` element and wrap it around it's parent

```xml
<XML>
    <BOX>
        <BLUR>
            <IMG>
                <URL value="./" />
            </IMG>
        </BLUR>
    </BOX>
</XML>
```

I have a semi working piece of code but its fairly messy and I wanted to see if there was a better way. Currently I am traversing down to the `IMG` element and checking its childElements for `BLUR`, but from that point I am curious about the best way to adjust the tree. Ideally I would prefer to avoid traversing and just run through the full document once but I assume that wouldn't play nice with modifying the iterators.

**Comment by renggli** [comment, marked as answer]
This works for me:

```dart
final actual = XmlDocument.parse('<XML><BOX><IMG><BLUR /><URL value="./" /></IMG></BOX></XML>');
final image = actual.findAllElements('IMG').single;
final blur = image.findAllElements('BLUR').single;
image.children.remove(blur);
image.replace(blur);
blur.children.add(image);
final expected = XmlDocument.parse('<XML><BOX><BLUR><IMG><URL value="./" /></IMG></BLUR></BOX></XML>');
expect(expected.toXmlString(), actual.toXmlString());
```
